src/test/java/org/sasanlabs/service/vulnerability/ssrf/SSRFVulnerabilityTest.java

```diff
@@ -0,0 +1,166 @@
+    }
+
+    @ParameterizedTest
+    @MethodSource("testParamsForLevel3")
+    void getVulnerablePayloadLevel3(String url, boolean isValid, String content)
+            throws IOException {
+        ResponseEntity<GenericVulnerabilityResponseBean<String>> responseEntity =
+                ssrfVulnerability.getVulnerablePayloadLevel3(url);
+        validateResponse(responseEntity, isValid, content);
+    }
+
+    private static Stream<Arguments> testParamsForLevel4() {
+        return Stream.of(
+                // Arguments: Input URL, Expected isValid response, Expected response body content
+                Arguments.of(INVALID_URL, false, INVALID_URL_MESSAGE),
+                Arguments.of(tempFileUrl, false, INVALID_URL_MESSAGE),
+                Arguments.of(METADATA_URL_AWS, false, INVALID_URL_MESSAGE),
+                Arguments.of(METADATA_URL_OTHER, false, INVALID_URL_MESSAGE),
+                Arguments.of(OTHER_URL, true, OTHER_URL_CONTENT),
+                Arguments.of(GIST_URL, true, GIST_URL_CONTENT));
+    }
+
+    @ParameterizedTest
+    @MethodSource("testParamsForLevel4")
+    void getVulnerablePayloadLevel4(String url, boolean isValid, String content)
+            throws IOException {
+        ResponseEntity<GenericVulnerabilityResponseBean<String>> responseEntity =
+                ssrfVulnerability.getVulnerablePayloadLevel4(url);
+        validateResponse(responseEntity, isValid, content);
+    }
+
+    private static Stream<Arguments> testParamsForLevel5() {
+        return Stream.of(
+                // Arguments: Input URL, Expected isValid response, Expected response body content
+                Arguments.of(INVALID_URL, false, INVALID_URL_MESSAGE),
+                Arguments.of(tempFileUrl, false, INVALID_URL_MESSAGE),
+                Arguments.of(METADATA_URL_AWS, false, INVALID_URL_MESSAGE),
+                Arguments.of(METADATA_URL_OTHER, false, INVALID_URL_MESSAGE),
+                Arguments.of(OTHER_URL, false, INVALID_URL_MESSAGE),
+                Arguments.of(GIST_URL, true, GIST_URL_CONTENT));
+    }
+
+    @ParameterizedTest
+    @MethodSource("testParamsForLevel5")
+    void getVulnerablePayloadLevel5(String url, boolean isValid, String content)
+            throws IOException {
+        ResponseEntity<GenericVulnerabilityResponseBean<String>> responseEntity =
+                ssrfVulnerability.getVulnerablePayloadLevel5(url);
+        validateResponse(responseEntity, isValid, content);
+    }
+
+    private void validateResponse(
+            ResponseEntity<GenericVulnerabilityResponseBean<String>> responseEntity,
+            boolean isValid,
+            String content) {
+        assertEquals(HttpStatus.OK, responseEntity.getStatusCode());
+        GenericVulnerabilityResponseBean<String> responseBody = responseEntity.getBody();
+        assertNotNull(responseBody);
+        assertEquals(isValid, responseBody.getIsValid());
+        assertEquals(content, responseBody.getContent());
```

Comment:
We may change the content on the gist url so instead of comparing entire Response, it is better to compare part of it as we might not remove project names in future.

Reply:
Updated to compare part of response.